Refactored navigation and header

helloimela opened the pull request:
**Describe pull-request**</br>
- Remove bug in IE by separating navigation & header so it is more simple
- Remove stickyfilljs
- Added all functionalities to the refactored solution


**Solving issue**</br>
Add which issue this pull-request solves by adding # plus the number of the issue (for example #123)
Fixes: #149 #166 


**Additional context**<br/>
Check few things below when reviewing:

- [x] make sure hamburger menu works on mobile view
- [x] make sure sticky navigation works 
- [x] symbol should stick as well
- [x] subnavigation behavior is correct

Commit:
Separate header and navigation

- Move navigation outside header
- Added store to watch for stuck in sticky nav

Commit:
Remove stuck from store

- Added behavior on mobile navigation
- Added sticky in IE
- Remove stuck from store
- Remove stickyfilljs dependencies

Commit:
Fix test error

- Removed navigation on header e2e
- Update navigation rendered element

Commit:
Fix IE error

- IE error 'clientWidth' of undefined because the calculation of nav width on didUpdate.
- Moved to didLoad

helloimela commented on src/components/navigation/navigation.e2e.ts:
```diff
@@ -44,7 +44,7 @@ describe('navigation', () => {
 
     // ensure ml-auto class is rendered
-    const linkSecondary = await page.findAll(`${root} .navbar-collapse:last-child a`);
+    const linkSecondary = await page.findAll(`${root} .navbar-collapse:last-of-type a`);
```
Now we have a symbol after navbar-collapse div, so, last-child is not working and we need to change it to last-of-type, which is a div.

Commit:
Removed unnecessary variable

Commit:
Fix sticky nav for IE

- Added custom margin and padding so it does not flicker on stickynav on IE fixed navigation
- Remove c-content from storybook template structure
- Do not display hamburger menu if navigation does not exist

Commit:
Remove uneccessary console log

Commit:
Fix navigation examples in storybook

- Removed overflow hidden in navigation from scania-theme
- Adjusment for the storybook examples regarding the changes

Commit:
Fixes for storybook

- reset navigation margin in storybook example

awinny reviewed (changes requested):
Look good but there are two things we need to fix:
- Scope margin-top 0 only on multi example pages
- Navigation prop sticky true/false set to true per default and false on multi example pages

Would also be good to change:
- In mobile mode margin-top for content can be overidden by css 0 !important instead of js

Commit:
Add fixes

- Scope margin 0 only for multiple examples
- Added public property for sticky option
- Disable sticky in multiple examples
- Set margin 0 in mobile from css instead of js
